src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -191,11 +194,124 @@ export class EthereumAdapter extends BaseProtocolAdapter {
   }
 
   override async fetchAvailableTokens(): Promise<IToken[]> {
```

Comment:
I suggest to catch errors in the body of this function or eventually in the place where it's executed.